package/src/nvidiamon.cpp

```diff
@@ -0,0 +1,207 @@
+
+  for (auto& stat : nvidia_stats) stat.second = 0;
+
+  auto cmd_result = prmon::cmd_pipe_output(cmd);
+  if (cmd_result.first) {
+    // Failed
+    return;
+  }
+
+  if (NVIDIA_DEBUG) {
+    std::cout << "nvidiamon::update_stats got the following output ("
+              << cmd_result.second.size() << "): " << std::endl;
+    int i = 0;
+    for (const auto& s : cmd_result.second) {
+      std::cout << i << " -> " << s << std::endl;
+      ++i;
+    }
+  }
+
+  // Loop over output
+  unsigned int gpu_idx{}, sm{}, mem{}, enc{}, dec{}, fb_mem{};
+  pid_t pid{};
+  std::string cg_type{}, cmd_name{};
+  std::unordered_map<unsigned int, bool>
+      activegpus{};  // Avoid double counting active GPUs
+  for (const auto& s : cmd_result.second) {
+    if (s[0] != '#') continue;
+    std::istringstream instr(s);
+    instr >> gpu_idx >> pid >> cg_type >> sm >> mem >> enc >> dec >> fb_mem >>
+        cmd_name;
+    auto read_ok = !(instr.fail() || instr.bad());  // eof() is ok
+    if (read_ok) {
+      if (NVIDIA_DEBUG) {
+        std::cout << "Good read: " << gpu_idx << " " << pid << " " << cg_type
+                  << " " << sm << " " << mem << " " << enc << " " << dec << " "
+                  << fb_mem << " " << cmd_name << std::endl;
+      }
+
+      // Filter on PID value, so we only add stats for our processes
+      for (auto const p : pids) {
+        if (p == pid) {
+          nvidia_stats["gpusmpct"] += sm;
+          nvidia_stats["gpumempct"] += mem;
+          nvidia_stats["gpufbmem"] += fb_mem;
+          if (!activegpus.count(gpu_idx)) {
+            ++nvidia_stats["ngpus"];
+            activegpus[gpu_idx] = true;
+          }
+        }
+      }
+    } else if (NVIDIA_DEBUG) {
+      std::clog << "Bad read of line: " << s << std::endl;
+      std::cout << "Parsed to: " << gpu_idx << " " << pid << " " << cg_type
+                << " " << sm << " " << mem << " " << enc << " " << dec << " "
+                << fb_mem << " " << cmd_name << std::endl;
+      std::cout << "StringStream status: good()=" << instr.good();
+      std::cout << " eof()=" << instr.eof();
+      std::cout << " fail()=" << instr.fail();
+      std::cout << " bad()=" << instr.bad() << std::endl;
+    }
+  }
+
+  if (NVIDIA_DEBUG) {
+    std::cout << "Parsed: " << nvidia_stats["ngpus"] << " "
+              << nvidia_stats["gpusmpct"] << " " << nvidia_stats["gpumempct"]
+              << " " << nvidia_stats["gpufbmem"] << std::endl;
+  }
+
+  // Update the statistics with the new snapshot values
+  ++iterations;
+  for (const auto& nvidia_param : nvidia_params) {
+    if (nvidia_stats[nvidia_param] > nvidia_peak_stats[nvidia_param]) {
+      nvidia_peak_stats[nvidia_param] = nvidia_stats[nvidia_param];
+    }
+    nvidia_total_stats[nvidia_param] += nvidia_stats[nvidia_param];
+    nvidia_average_stats[nvidia_param] =
+        double(nvidia_total_stats[nvidia_param]) / iterations;
+  }
+}
+
+// Return the nvidiaers
```

Comment:
Is there a typo here?

Reply:
I have no idea what that was supposed to be!